cast/src/lib.rs

```diff
@@ -667,11 +667,14 @@ impl SimpleCast {
         let (contract_abis, contract_names): (Vec<Abi>, Vec<String>) = match address_or_path {
             InterfacePath::Local(path) => {
                 let file = std::fs::read_to_string(&path).wrap_err("unable to read abi file")?;
-                (
-                    vec![serde_json::from_str(&file)
-                        .wrap_err("unable to parse json ABI from file")?],
-                    vec!["Interface".to_owned()],
-                )
+
+                let mut json: serde_json::Value = serde_json::from_str(&file)?;
+                let json = if !json["abi"].is_null() { json.take("abi") } else { json };
```

Comment:
It seems it was meant to be like this, my bad :sweat_smile: 

```suggestion
                let json = if !json["abi"].is_null() { json["abi"].take() } else { json };
```